iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AreaImpl.java

```diff
@@ -27,16 +28,19 @@ public class AreaImpl extends AbstractIdentifiable<Area> implements Area {
     private final List<AreaBoundary> areaBoundaries;
     private final Set<VoltageLevel> voltagelevels;
 
+    protected boolean removed = false;
+
     private final class AreaListener extends DefaultNetworkListener {
         @Override
-        public void beforeRemoval(Identifiable identifiable) {
+        public void beforeRemoval(Identifiable<?> identifiable) {
             if (identifiable instanceof DanglingLine danglingLine) {
                 // if dangling line removed from network, remove its boundary from this extension
                 AreaImpl.this.removeAreaBoundary(danglingLine.getBoundary());
             } else if (identifiable instanceof Connectable<?> connectable) {
                 // if connectable removed from network, remove its terminals from this extension
                 connectable.getTerminals().forEach(AreaImpl.this::removeAreaBoundary);
             }
+            // If the identifiable removed is a VoltageLevel: the area voltageLevels attribute will be updated by VoltageLevel::remove()
```

Comment:
I find this comment difficult to understand. Maybe something like this will be clearer:
```suggestion
            // When a VoltageLevel is removed, its areas' voltageLevels attributes are directly updated. This is not managed via this listener.
```